airbyte-workers/src/main/java/io/airbyte/workers/RecordSchemaValidator.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright (c) 2021 Airbyte, Inc., all rights reserved.
+ */
+
+package io.airbyte.workers;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import io.airbyte.config.StandardSyncInput;
+import io.airbyte.protocol.models.AirbyteMessage;
+import io.airbyte.protocol.models.ConfiguredAirbyteStream;
+import io.airbyte.validation.json.JsonSchemaValidator;
+import io.airbyte.validation.json.JsonValidationException;
+
+public class RecordSchemaValidator {
+
+  public void validateSchema(final AirbyteMessage message, final StandardSyncInput syncInput) throws RecordSchemaValidationException {
```

Comment:
I think it would slightly simplify this interface if this took in an `AirbyteRecordMessage` rather than an `AirbyteMessage`. This is also helpful as it indicates that we only expect to pass _record_ messages to this class